docs/docs/markers.mdx

```diff
@@ -0,0 +1,302 @@
+
+By default, the script will validate the marker definitions against your bot's `domain.yml` file. To specify a different domain file, use the optional `--domain` argument.
+
+By default, the script will retrieve the tracker store specified in your bot's `endpoint.yml` file. However, you can specify a different endpoint file using the optional `--endpoint` argument.
+
+The script provides three tracker loading strategies: `all`, `sample_n`, and `first_n`. The option `all` will process all the trackers in your tracker store. The other two strategies process a subset of `N` trackers from all trackers, either sequentially (by using `first_n`), or by sampling uniformly without replacement (by using `sample_n`). The sampling strategy also allows you to set the random seed.
+For more information on the usage of each strategy, type the following:
+```bash
+rasa evaluate markers <strategy> --help
+```
+replacing `<strategy>` with one of: `all`, `first_n`, and `sample_n`.
+
+Each tracker in the tracker store can contain multiple sessions. The script will process each session separately, indexing them by `session_idx`.
+
+The script above will output the extracted markers in the specified output file: `extracted_markers.csv`. It will also compute overall summary statistics and per-session summary statistics and output them in the files `stats-overall.csv` and `stats-per-session.csv`.
+You can change prefix `stats` in the file names using the optional argument `--stats-file-prefix `.
+For example, the following script will produce the files: `marker-statistics-overall.csv` and `marker-statistics-per-session.csv`:
+```bash
+rasa evaluate markers all --stats-file-prefix "marker-statistics" extracted_markers.csv
+```
+
+
+The next two sections give more detail about the extracted markers and computed statistics.
+
+
+### Extracted Markers
+
+For each marker defined in the marker configuration file, the following information is extracted:
+1. The number of user turns preceding the event at which a marker applied. Each `UserUttered` event is treated as a user turn.
+2. The index of the event at which a marker applied.
+
+For the `at_least_once` marker, the information above will only be extracted for the first occurrence.
+
+The extracted markers will be output to the `.csv` file you specify in the script, for example, `extracted_markers.csv` (in the script above).
+The extracted markers output file contains the following columns:
+
+- `sender_id`: taken from the trackers.
+- `session_idx`: an integer indexing sessions, starting with `0`.
+- `marker`: the unique identifier you specified for the marker.
+- `event_idx`: an integer indexing events, starting with `0`.
+- `num_preceding_user_turns`: an integer specifying the number of user turns which preceded the event at which the marker applied.
+
+Here is a sample of the extracted markers output file:
+
+```
+sender_id,session_idx,marker_name,event_id,num_preceding_user_turns
```

Comment:
@usc-m and @ka-bu I noticed that we use marker_name in the extracted marker output, but use marker in the stats files. I would like to make it consistent if that's okay, switching to marker for both.

Not sure if I should do it in this PR or in another one (possibly combining it with any CLI changes). What do you think?

Reply:
Sounds good - I'd split it out into it's own PR to not clutter this one too much (or combine with the CLI changes)